src/huggingface_hub/hf_api.py

```diff
@@ -9525,6 +9529,51 @@ def list_user_following(self, username: str) -> Iterable[User]:
         for followed_user in r.json():
             yield User(**followed_user)
 
+    def auth_check(self, repo_id: str, token: Union[bool, str, None] = None) -> None:
+    """
+    Check if the provided user token has access to a specific repository on the Hugging Face Hub.
+
+    Args:
+        repo_id (`str`):
+            The repository to like. Example: `"user/my-cool-model"`.
+
+        token (Union[bool, str, None], optional):
+            A valid user access token (string). Defaults to the locally saved
+            token, which is the recommended method for authentication (see
+            https://huggingface.co/docs/huggingface_hub/quick-start#authentication).
+            To disable authentication, pass `False`.
+
+    Returns:
+        `None`: This method does not return any value. It will either complete successfully or raise an exception if access is denied.
+
+    Raises:
+        [`~utils.RepositoryNotFoundError`]:
+            If repository is not found (error 404): wrong repo_id/repo_type, private
+            but not authenticated or repo does not exist.
+
+        [`~utils.GatedRepoError`]
+            If the repository exists but is gated and the user is not on the authorized
+            list.
+
+    Example:
+        >>> api = HfApi(token="your_token")
+        >>> api.auth_check(repo_id="user/my-cool-model")
```

Comment:
```suggestion
    Example:
    ```py
        >>> from huggingface_hub import HfApi
        >>> api = HfApi(token="your_token")
        >>> api.auth_check(repo_id="user/my-cool-model")
    ```
```